sw/device/lib/testing/entropy_testutils.c

```diff
@@ -31,31 +32,93 @@ static void setup_csrng(void) {
   dif_csrng_t csrng;
   CHECK_DIF_OK(dif_csrng_init(
       mmio_region_from_addr(TOP_EARLGREY_CSRNG_BASE_ADDR), &csrng));
+
+  CHECK_DIF_OK(dif_csrng_stop(&csrng));
   CHECK_DIF_OK(dif_csrng_configure(&csrng));
 }
 
-static void setup_edn(void) {
+static void setup_edn(bool auto_mode) {
   // Temporary solution to configure/enable the EDN and CSRNG to allow OTBN to
   // run before a DIF is available,
   // https://github.com/lowRISC/opentitan/issues/6082
-  uint32_t reg =
-      bitfield_field32_write(0, EDN_CTRL_EDN_ENABLE_FIELD, kMultiBitBool4True);
+  // disable edn.
+  uint32_t reg = 0;
+  reg =
+      bitfield_field32_write(0, EDN_CTRL_EDN_ENABLE_FIELD, kMultiBitBool4False);
   reg = bitfield_field32_write(reg, EDN_CTRL_BOOT_REQ_MODE_FIELD,
-                               kMultiBitBool4True);
+                               kMultiBitBool4False);
   reg = bitfield_field32_write(reg, EDN_CTRL_AUTO_REQ_MODE_FIELD,
                                kMultiBitBool4False);
   reg = bitfield_field32_write(reg, EDN_CTRL_CMD_FIFO_RST_FIELD,
                                kMultiBitBool4False);
   mmio_region_write32(mmio_region_from_addr(TOP_EARLGREY_EDN0_BASE_ADDR),
                       EDN_CTRL_REG_OFFSET, reg);
-  mmio_region_write32(mmio_region_from_addr(TOP_EARLGREY_EDN1_BASE_ADDR),
-                      EDN_CTRL_REG_OFFSET, reg);
+
```

Comment:
@timothytrippel 
this really needs to be cleaned up still...but i've added it here temporarily for now.

Reply:
no problem, once this merges I will clean up the entropy testutils.